### DISCUSSION TITLE
v7 Manual (server-side) group by with pagination

### DISCUSSION THREAD

**Discussion by jonathan-koskas-shift** [TanStack/table#3988] [Q&A]
Hi all !
First of all I want to thanks for this awesome library !

I'm trying to build a full server-side controlled react-table with group by, filtering, sorting, pagination etc. But I definitly can not find any example on how to achieve manual group by server-side, I cannot figure how to provider already grouped data to the table hooks ( I tried with subRows key but it appears the group by state is not working with isGrouped, isExpanded etc.)

Can someone show me an example somewhere ? 

Thanks a lot !

**Comment by Beguface** [comment]
I am looking for the same :(

**Comment by MartaMGD** [comment]
I have the same problem!

**Comment by santanapaulo** [comment]
@tannerlinsley can you help us here?

**Comment by dilfin07** [comment]
someone was able to find any implementation for server-side grouping for the implementation of the tree?

**Comment by tkfmed** [comment]
It's a bit annoying that something what's potentially possible it's not documented. What do you mean by writing [here](https://tanstack.com/table/v8/docs/api/features/grouping#manualgrouping) `will expect you to manually group the rows before passing them to the table`? What do you mean by writing `manually`? Where should I transform the data?
@tannerlinsley

**Reply by devi-prsd** [reply]
https://tanstack.com/table/v8/docs/api/features/grouping#ongroupingchange this also important, basically when you set it to manual, any interactions with the table will call this function that will provide you with a state. You can do some grouping locally or on the server and return a new of data that the table will display
